test/integration/node_pool/controls/gcloud.rb

```diff
@@ -0,0 +1,168 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+project_id = attribute('project_id')
+location = attribute('location')
+cluster_name = attribute('cluster_name')
+
+credentials_path = attribute('credentials_path')
+ENV['CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE'] = credentials_path
+
+control "gcloud" do
+  title "Google Compute Engine GKE configuration"
+  describe command("gcloud --project=#{project_id} container clusters --zone=#{location} describe #{cluster_name} --format=json") do
+    its(:exit_status) { should eq 0 }
+    its(:stderr) { should eq '' }
+
+    let!(:data) do
+      if subject.exit_status == 0
+        JSON.parse(subject.stdout)
+      else
+        {}
+      end
+    end
+
+    describe "default node pool" do
+      let(:default_node_pool) { data['nodePools'].select { |p| p['name'] == "default-pool" }.first }
```

Comment:
Possible target for `include` / `including`.

Reply:
Is something like this what you're getting at?

```ruby
    describe "default node pool" do
      let(:node_pools) { data['nodePools'].map(&:to_h) }
      let(:default_node_pool) { data['nodePools'].select { |p| p['name'] == "default-pool" }.first }

      it "exists" do
        expect(node_pools).to include(
          including(
            "name" => "default-pool",
          )
        )
      end

      it "has no initial node count" do
        expect(default_node_pool['initialNodeCount']).to eq nil
      end

      it "does not have autoscaling enabled" do
        expect(default_node_pool['autoscaling']).to eq nil
      end
    end
```

In this case, the principal difference between what already exists and what I've pasted above is an explicit example case that asserts there is a node pool named `default-pool`. A failure would happen either way, but the error message as currently written would admittedly require a bit more investigation to reach the conclusion that `data['nodePools'].select { |p| p['name'] == "default-pool" } == []`, and that `#first` had been called on the empty array. I'm all for providing clear error messages -- just want to make sure this is what you're getting at.

Reply:
Yup, but we could go even further in this case since all three examples are really dealing with the same hash:

```ruby
describe "default node pool" do
  it "has no initial node count or autoscaling" do
    expect(node_pools).to include(
      including(
        "name" => "default-pool",
        "initialNodeCount" => nil,
        "autoscaling" => nil
      )
    )
  end
end
```

A failure of this example should indicate if the "default-pool" element is missing or if it violates one of the other constraints. It may be preferable to leave the cases as discrete examples, but even still, we can use the `include` / `including` matcher to avoid the out of band filtering of the node pools.

Reply:
Ah, that's a fair point. Thanks for clarifying!